tobac/segmentation.py

```diff
@@ -986,6 +994,9 @@ def segmentation_timestep(
 
     # Finished PBC checks and new PBC updated segmentation now in segmentation_mask.
     # Write resulting mask into cube for output
+    segmentation_mask_cpy = copy.deepcopy(segmentation_mask)
+    segmentation_mask[segmentation_mask_cpy == 0] = above_threshold_unseg_num
+    segmentation_mask[segmentation_mask_cpy == -1] = below_threshold_num
```

Comment:
**Suggestion:** Rather than copying the entire array, we could instead save the masks first. This may save some memory as the boolean arrays should use less

```suggestion
    wh_below_threshold = segmentation_mask == -1
    wh_unsegmented = segmentation_mask == 0
    segmentation_mask[wh_unsegmented] = above_threshold_unseg_num
    segmentation_mask[wh_below_threshold] = below_threshold_num
```

Reply:
Good point. I've switched to bool arrays.